table_foosball/database.rb

```diff
@@ -0,0 +1,117 @@
+require_relative "./text_tools"
+
+class Database
+  include TextTools
+
+  def initialize(name)
+    @name = name
+    @filename = "./#{name}.sdb"
+  end
+
+  def clear
+    File.truncate("./#{@name}", 0)
+  end
+
+  def save(objs)
+    file = File.open(@filename, "r+")    
+    file.rewind
```

Comment:
So this is kind of interesting. Can you explain what the difference between `File.open(@filename, "w")` and `File.open(@filename, "r+); file.rewind` is?